examples/discovery-search-app/cypress/integration/collection_filter/spec.ts

```diff
@@ -0,0 +1,142 @@
+            'postQuerySingleCollection'
+          );
+          cy.get('.bx--list-box__menu-item')
+            .contains('finnegans wake')
+            .click();
+          cy.wait('@postQuerySingleCollection').as('singleCollectionQueryObject');
+        });
+
+        it('should make a query against only the selected collection', () => {
+          cy.get('@singleCollectionQueryObject')
+            //@ts-ignore
+            .its('requestBody.collection_ids')
+            .should('contain', 'paris19221939');
+        });
+
+        it('the clear all selected collections button appears', () => {
+          cy.get('div[aria-label="Clear Selection"]').should('exist');
+        });
+
+        describe('and we click the clear selected collections button', () => {
+          beforeEach(() => {
+            cy.route('POST', '**/query?version=2019-01-01', '@queryJSON').as('postQuery');
+            cy.get('div[aria-label="Clear Selection"]').click();
+            cy.wait('@postQuery').as('clearedCollectionsQueryObject');
+          });
+
+          it('makes a query against all available collections', () => {
+            cy.get('@clearedCollectionsQueryObject')
+              //@ts-ignore
+              .its('requestBody.collection_ids')
+              .should('be.empty');
+          });
+        });
+
+        describe('and we select another collection', () => {
+          beforeEach(() => {
+            cy.route('POST', '**/query?version=2019-01-01', 'doubleCollectionQueryJSON').as(
+              'postQueryDoubleCollection'
+            );
+            cy.get('.bx--list-box__menu-item')
+              .contains('deadspin')
+              .click();
+            cy.wait('@postQueryDoubleCollection').as('doubleCollectionQueryObject');
+          });
+
+          it('makes a query against both of the selected collections', () => {
+            cy.get('@doubleCollectionQueryObject')
+              //@ts-ignore
+              .its('requestBody.collection_ids')
+              .should('contain', 'deadspin9876')
+              .and('contain', 'paris19221939');
+          });
+
+          describe('and we clear the selected collections', () => {
+            beforeEach(() => {
+              cy.route('POST', '**/query?version=2019-01-01', '@queryJSON').as('postQuery');
+              cy.get('div[aria-label="Clear Selection"]').click();
+              cy.wait('@postQuery').as('originalQueryObject');
+            });
+
+            it('makes a query against all available collections', () => {
+              cy.get('@originalQueryObject')
+                //@ts-ignore
+                .its('requestBody.collection_ids')
+                .should('be.empty');
+            });
+          });
+        });
+      });
+
+      describe('and we click the selection box again', () => {
+        beforeEach(() => {
+          cy.get('#collection-facet-select').click();
+        });
+
+        it('the collection filter dropdown disappears', () => {
```

Comment:
Not sure if this is too verbose and we just want to trust that carbon will take care of this for us. But perhaps putting in a test here to make sure if a click even happens outside of the collection filter dropdown. that the dropdown also disappears. I only bring this up because I remember seeing bugs like this in tooling that went uncaught

Reply:
Yeah, I could see this being too much detail. I don't have a strong feeling, but I'm leaning towards leaving it here. Maybe something to get rid of if we start breaking our tests down into smaller files.

Reply:
Before I was wondering if we needed a test to cover that the collections facet closes when a click event happens outside of the facet component, because I've seen this error make it's way into Tooling. Probably not important enough to add if this isn't as simple as copy and paste this test, and then put the click event else where